### DISCUSSION TITLE
layout is messy for certain pdf

### DISCUSSION THREAD

**Discussion by Yichen-fqyd** [pymupdf/PyMuPDF#1149] [Looking for help]
Hi,

I am converting the pdf to html, but for some files the layout came very messy, is it possible to fix that? and what might be the cause of this?

Thank you so much,
attached example file

[16-18 AIG Loss Runs.pdf](https://github.com/pymupdf/PyMuPDF/files/6831875/16-18.AIG.Loss.Runs.pdf)

**Comment by JorjMcKie** [comment]
(X)HTML and XML output are wrappers of MuPDF functions. I (or resp. PyMuPDF) can do nothing about them.
Disappointing appearances are mostly caused by the text fonts: the HTML source contains references to the font **_names_**, but does not contain the font **_files_** - as opposed to the PDF situation, where the font **_files are embedded_** in the PDF.
So the chosen browser decides which font to actualy use for building the appearance. Some browsers also have their own fonts which they employ when encountering a matching fontname in the HTML source etc.
You can e.g. check how different the same HTML source looks with different browsers.
